compiler/src/dotty/tools/dotc/transform/TreeChecker.scala

```diff
@@ -122,6 +122,10 @@ class TreeChecker extends Phase with SymTransformer {
     val squahsedPhase = ctx.squashed(prevPhase)
     ctx.echo(s"checking ${ctx.compilationUnit} after phase ${squahsedPhase}")
 
+    if (ctx.isAfterTyper) {
```

Comment:
@DarkDimius is this the right place to invoke the new check? Do I need to pass a fresh context?